File: dashboard/src/models/correlation.ts
export interface TelemetryPoint {
  timestamp: string
  tag: string
  value: number
}

export interface CorrelationDetail {
  pattern: string
  base_severity: string
  final_severity: string
  network_summary: string | null
  process_summary: string | null
  link_summary: string | null
}

export interface CorrelationContext {
  alert_id: number
  alert_timestamp: string
  rule_id: string | null
  severity: string
  source: string
  destination: string
  correlated: boolean
  window_start: string
  window_end: string
  telemetry: TelemetryPoint[]
  correlation: CorrelationDetail | null
}

File: edge/domain/models/correlation.py
from dataclasses import dataclass
from datetime import datetime


@dataclass
class Correlation:
    event_id: int
    pattern: str
    base_severity: str
    final_severity: str
    network_summary: str | None
    process_summary: str | None
    link_summary: str | None
    created_at: datetime
    id: int | None = None